x-pack/plugins/global_search/common/types.ts

```diff
@@ -63,6 +63,8 @@ export interface GlobalSearchProviderResult {
   score: number;
   /** an optional record of metadata for this result */
   meta?: Record<string, Serializable>;
+  /** an array of tag IDs associated with this result */
+  tags: string[];
 }
```

Comment:
These are optional meta that may not be present on every kind of results. Just add a `tags` or `tagIds` entry to `meta` instead, this is what this record was meant to be used for 😄